### ISSUE TITLE
Clickhouse version > 22.3 fails to run

### ISSUE THREAD

**Issue by sandeeppnaik** [Altinity/clickhouse-operator#1006]
HI team, 

I was running some of the sample CRs from [here](https://github.com/Altinity/clickhouse-operator/tree/master/docs/chi-examples).
They all work great, but for when I run the same examples with a clickhouse version > 22.3 e.g. 22.7, 22.8.4.7 etc., the pods fail to start.

I cleaned up all my custom resources and then installed a new CR.

Observation
- The CHI is reporting errors while performing reconciliation
- When I shell into the pod to see the logs of what the manual run of `clickhouse-server` return, the app seems to be stuck even before the logging is triggered. 

Sample CR that fails:

```apiVersion: "clickhouse.altinity.com/v1"
kind: "ClickHouseInstallation"
metadata:
  name: "vers-update"
spec:
  configuration:
    clusters:
      - name: rep
        templates:
          podTemplate: clickhouse:22.8
        layout:
          shards:
            - replicas:
                - name: n1
                - name: n2
                - name: n3

  templates:
    podTemplates:
      - name: clickhouse:22.8
        spec:
          containers:
            - name: clickhouse-pod
              image: clickhouse/clickhouse-server:22.8
```

Here is the log from the CRD

```
action: Create StatefulSet clickhouse-cluster/chi-vers-update-rep-n2 - started                                                                                           │
actions:                                                                                                                                                                 │
- 2022-09-12T20:36:35.45317811Z Create StatefulSet clickhouse-cluster/chi-vers-update-rep-n2                                                                             │
  - started                                                                                                                                                              │
- 2022-09-12T20:36:34.653371311Z Create ConfigMap clickhouse-cluster/chi-vers-update-deploy-confd-rep-n2                                                                 │
- 2022-09-12T20:36:33.65455401Z Reconcile Host n2 started                                                                                                                │
- 2022-09-12T20:36:32.997450613Z Reconcile Host n1 completed                                                                                                             │
- 2022-09-12T20:36:31.452441365Z Update ConfigMap clickhouse-cluster/chi-vers-update-common-configd                                                                      │
- 2022-09-12T20:36:30.451350978Z Adding tables on shard/host:0/0 cluster:rep                                                                                             │
- 2022-09-12T20:36:29.662059118Z Create Service clickhouse-cluster/chi-vers-update-rep-n1                                                                                │
- '2022-09-12T20:36:28.550753417Z FAILED to reconcile StatefulSet: chi-vers-update-rep-n1                                                                                │
  CHI: vers-update '                                                                                                                                                     │
- 2022-09-12T20:36:28.500931384Z Create StatefulSet clickhouse-cluster/chi-vers-update-rep-n1                                                                            │
  - error ignored                                                                                                                                                        │
- 2022-09-12T20:31:21.71312086Z Create StatefulSet clickhouse-cluster/chi-vers-update-rep-n1                                                                             │
  - started                                                                                                                                                              │
- 2022-09-12T20:31:20.891673104Z Create ConfigMap clickhouse-cluster/chi-vers-update-deploy-confd-rep-n1                                                                 │
- 2022-09-12T20:31:19.888872898Z Reconcile Host n1 started                                                                                                               │
- 2022-09-12T20:31:19.092231993Z Create ConfigMap clickhouse-cluster/chi-vers-update-common-usersd                                                                       │
- 2022-09-12T20:31:18.090567024Z Create ConfigMap clickhouse-cluster/chi-vers-update-common-configd                                                                      │
- 2022-09-12T20:31:17.097935541Z Create Service clickhouse-cluster/clickhouse-vers-update                                                                                │
- 2022-09-12T20:31:15.968933943Z reconcile started                                                                                                                       │
added: 1                                                                                                                                                                 │
chop-commit: d7e654a                                                                                                                                                     │
chop-date: 2022-08-25T07:42:36                                                                                                                                           │
chop-ip: 10.21.114.28                                                                                                                                                    │
chop-version: 0.19.1                                                                                                                                                     │
clusters: 1                                                                                                                                                              │
endpoint: clickhouse-vers-update.clickhouse-cluster.svc.cluster.local                                                                                                    │
error: 'FAILED to reconcile StatefulSet: chi-vers-update-rep-n1 CHI: vers-update '                                                                                       │
errors:                                                                                                                                                                  │
- '2022-09-12T20:36:28.550753417Z FAILED to reconcile StatefulSet: chi-vers-update-rep-n1                                                                                │
  CHI: vers-update '
```

I am wondering why this setup works seamlessly for version <= 22.3.

**Comment by Slach** [comment]
could you share pod logs?
```bash
kubectl logs -n clickhouse-cluster pod/chi-vers-update-rep-n1-0   --since=24h
```

Moreover, could you install not released yet 0.20.0 operator? 
```bash
kubectl apply -f https://github.com/Altinity/clickhouse-operator/blob/0.20.0/deploy/operator/clickhouse-operator-install-bundle.yaml
```

**Comment by sandeeppnaik** [comment]
Hey @Slach ! Upgrading the operator to the latest has resolved the problem !!
Now the nodes upgrade fine, also roll-out of version increment works great. Thanks for the suggestion 🙏 
I was wanted to investigate the logs, however, there were no logs coming through while this was occurring. I'l close this now.
